### PR TITLE
fix eager loading extension with many to many

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ∅
| License       | MIT
| Doc PR        | ∅

Small fix when joinColumns does not exist. Also fixes edge cases where the joinTable does not exist (case of a custom association table).

### CODE REVIEW

**Comment by dunglas** [comment]
👍

**Comment by teohhanhui** [comment]
Any tests?

**Comment by soyuka** [comment]
Added tests.

File: src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php [hunk 1]
```diff
@@ -80,7 +80,8 @@ private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass
                 continue;
             }
 
-            $method = false === $mapping['joinColumns'][0]['nullable'] ? 'innerJoin' : 'leftJoin';
+            $joinColumns = $mapping['joinColumns'] ?? $mapping['joinTable']['joinColumns'] ?? null;
+            $method = null === $joinColumns ? 'leftJoin' : (false === $joinColumns[0]['nullable'] ? 'innerJoin' : 'leftJoin');
```
Comment (dunglas) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: Nested ternary operators are very hard to read. Can you change change it to the following?

```php
$joinColumns = $mapping['joinColumns'] ?? $mapping['joinTable']['joinColumns'] ?? null;
if (null === $joinColumns) {
    $method = 'leftJoin';
} else {
    $method = false === $joinColumns[0]['nullable'] ? 'innerJoin' : 'leftJoin';
}
```

Thanks!

Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: Oh you think? I find it pretty and easy to understand.

Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: Can I keep the first one?

```
$joinColumns = $mapping['joinColumns'] ?? $mapping['joinTable']['joinColumns'] ?? null;

if (null === $joinColumns) {
  $method = 'leftJoin';
} else {
 $method = false === $joinColumns[0]['nullable'] ? 'innerJoin' : 'leftJoin';
}
```

**Comment by dunglas** [comment]
@soyuka can you fix CS issues and rebase please?

**Comment by soyuka** [comment]
done

**Comment by dunglas** [comment]
Thanks @soyuka
